src/ray/rpc/raylet/raylet_server.h

```diff
@@ -0,0 +1,258 @@
+#ifndef RAY_RPC_RAYLET_SERVER_H
+#define RAY_RPC_RAYLET_SERVER_H
+
+#include "src/ray/rpc/grpc_server.h"
+#include "src/ray/rpc/server_call.h"
+
+#include "src/ray/protobuf/raylet.grpc.pb.h"
+#include "src/ray/protobuf/raylet.pb.h"
+
+namespace ray {
+namespace rpc {
+
+/// Implementations of the `RayletService`, check interface in
+/// `src/ray/protobuf/raylet.proto`.
+class RayletServiceHandler {
+ public:
+  virtual void HandleRegisterClientRequest(const RegisterClientRequest &request,
```

Comment:
A lot of these handlers have to call `send_reply_callback` with the same arguments at the end of the function, which feels brittle. Do you think we can have a wrapper function to call the `send_reply_callback` for handlers that fit this pattern? I'm okay with doing this in a future PR.

Reply:
You mean a wrapper function like `ReplyOk(const SendReplyCallback &send_reply_callback) { send_reply_callback(Stauts::OK(), nullptr, nullptr); }`?

Reply:
Yes, something like that, just so we don't have to add `send_reply_callback` to the signature of functions like `HandleRegisterClientRequest` every time.

Reply:
I think `send_reply_callback` needs to be added to the signature. because it's a lambda passed from `HandleRequestImpl`. It captures some local information about this request.

Reply:
Right, I'm suggesting that we remove `send_reply_callback` from the signature of `HandleRegisterClientRequest` and wrap `HandleRegisterClientRequest` in a lambda that calls `send_reply_clalback`. So something like this:
```
auto wrapped_handler = [](const rpc::RegisterClientRequest &request,
                                           rpc::RegisterClientReply *reply,
                                           rpc::SendReplyCallback send_reply_callback) {
  HandleRegisterClientRequest(request);
  send_reply_callback(Status::OK(), nullptr, nullptr);
};
```
We can probably create a templated helper function that returns the wrapped handler.